### DISCUSSION TITLE
[Feature Request] Support CSRNDArray in kv_store

### DISCUSSION THREAD

**Discussion by zerolxf** [apache/incubator-mxnet#13381] [Q&A]
## Description
In distributed training, communication takes up a lot of time. In order to compress the communication time, an optimization method is to make the gradient sparse.

But for now kv_store doesn't support push CSRNDArray values.
Although support for this feature may cause other problems, right?

For example, for CSRNDArray format

>The format is designed for 2D matrices with a large number of columns, and each row is sparse

But kv_store probably transfers multidimensional arrays

**Comment by vdantu** [comment]
@mxnet-label-bot add [KVStore, Feature request, Question]

**Comment by eric-haibin-lin** [comment]
Yes! I am also thinking about doing this. I am curious - what's your use case?

**Comment by zerolxf** [comment]
I want to implement the optimizer described in the paper "Sparse Communication for Distributed Gradient Descent" , available at https://arxiv.org/abs/1704.05021

I tried to modify the value of NDArray directly through index, but it was a bit slow.
